### ISSUE TITLE
getItem() return type always contains null

### ISSUE THREAD

**Issue by dxptqhtlutehvlyxcmtg** [ciriousjoker/ts-localstorage#5]
Thanks for this library.

Consider the following usage:

```typescript
const key = new LocalKey<Options>("mykey", { id: 'abc' }, {
  hasDefaultValue: true,
});

const savedOptions = LocalStorage.getItem(key);
```

Even though in this circumstance `getItem()` will never return `null`, we can't use `savedOptions` without checking for `null` first.

For example, the following function won't compile:
```typescript

// TS error "null is not assignable to Options"
function getSavedOptions():Options {
   return LocalStorage.getItem(key);
}
```

We could use `LocalStorage.getItem(key) ?? { id: 'abc' }`, but this defeats the purpose of providing a default in the first place.

**Comment by ciriousjoker** [comment]
Yes, this can not easily be fixed.

**Solution 1:**
The Typescript type checker can't know if you have set hasDefaultValue to true and using Options as the return type might be dangerous. Solving this would require a split approach, like LocalKey for non-nullable keys with defaults and LocalKeyNullable for nullable keys. This would be a major breaking change and it would make this a lot more complicated with very little benefit imo.

**Solution 2:**
Maybe a second optional type parameter could be added as a sort of typescript variable to determine if the thing is nullable? That could work, in that case you'd need to write `LocalKey<Options, Options>`. The second Options would be the type of the default value returned if the first type isn't set. Maybe it's even possible to infer this. This would definitely be the best case solution here and the one I'd try first.

Either way, I have somewhat limited time right now, not sure I'll get to it in the near future. I'm open to PRs though.

For now, you could use exclamation marks and the linter will tell you to remove them once a proper solution is available.

**Comment by ciriousjoker** [comment]
@dxptqhtlutehvlyxcmtg This has been fixed with `v3.1.0`
